core/templates/hashfuncs.h

```diff
@@ -100,7 +100,73 @@ static inline uint32_t hash_one_uint64(const uint64_t p_int) {
 	return uint32_t(v);
 }
 
-static inline uint32_t hash_djb2_one_float(double p_in, uint32_t p_prev = 5381) {
+// Murmurhash3 32-bit version.
+// All MurmurHash versions are public domain software, and the author disclaims all copyright to their code.
+
+static _FORCE_INLINE_ uint32_t rotl32(uint32_t x, int8_t r) {
+	return (x << r) | (x >> (32 - r));
+}
+
+static _FORCE_INLINE_ uint32_t fmix32(uint32_t h) {
+	h ^= h >> 16;
+	h *= 0x85ebca6b;
+	h ^= h >> 13;
+	h *= 0xc2b2ae35;
+	h ^= h >> 16;
+
+	return h;
+}
+
+static _FORCE_INLINE_ uint32_t hash_murmur3_32(const void *key, int len) {
```

Comment:
I would probably make this compatible with the djb2 call style, where you can call the function multiple times and if you dont pass a second argument, the seed is used.

The reason is that, many times, things that need to be hashed do not appear in the form of buffers.

Reply:
The beauty of this hash function is that it is a general purpose hash function that just works on any form of data. As it is already done with several types in this PR, you can pass any type together with its size in bytes to it. For example Vector2i: `hash_murmur3_32_buffer(&p_vec, sizeof(Vector2i));`
I had dedicated versions for primitive types, but they didn't provide any performance benefit compared to this general variant. Nearly all places where dbj2 is used can be easily replaced with this buffer version. (However, at one point, I had replaced the dbj2 string hash functions (in ustring.cpp) with murmur3 but that broke ClassDB for some reason and I have no clue why).

Besides that, I am not sure if the quality of murmur3 will still be the same if I modify it to work like the dbj2 hash functions (because that is not how it was intended to be used and hash functions are quite fragile). I could do some tests, but that might take some time.
For now I added a seed parameter which defaults to the same random prime number as before.